src/databricks/labs/ucx/azure/azure_credentials.py

```diff
@@ -0,0 +1,283 @@
+            directory_id=sp_migration.service_principal.directory_id,
+            application_id=sp_migration.service_principal.client_id,
+            client_secret=sp_migration.client_secret,
+        )
+        comment = f"Created by UCX during migration to UC using Azure Service Principal: {sp_migration.service_principal.principal}"
+        read_only = False
+        if sp_migration.service_principal.privilege == Privilege.READ_FILES.value:
+            read_only = True
+        # create the storage credential
+        storage_credential = self._ws.storage_credentials.create(
+            name=name, azure_service_principal=azure_service_principal, comment=comment, read_only=read_only
+        )
+
+        validation_result = self._validate_storage_credential(
+            storage_credential, sp_migration.service_principal.prefix, read_only
+        )
+        return validation_result
+
+    def _validate_storage_credential(
+        self, storage_credential, location: str, read_only: bool
+    ) -> StorageCredentialValidationResult:
+        # storage_credential validation creates a temp UC external location, which cannot overlap with
+        # existing UC external locations. So add a sub folder to the validation location just in case
+        try:
+            validation = self._ws.storage_credentials.validate(
+                storage_credential_name=storage_credential.name, url=location, read_only=read_only
+            )
+            return StorageCredentialValidationResult.from_storage_credential_validation(storage_credential, validation)
+        except InvalidParameterValue:
+            logger.warning(
+                "There is an existing external location overlaps with the prefix that is mapped to the service principal and used for validating the migrated storage credential. Skip the validation"
+            )
+            return StorageCredentialValidationResult.from_storage_credential_validation(
+                storage_credential,
+                ValidateStorageCredentialResponse(
+                    is_dir=None,
+                    results=[
+                        ValidationResult(
+                            message="The validation is skipped because an existing external location overlaps with the location used for validation."
+                        )
+                    ],
+                ),
+            )
+
+    def execute_migration(self, prompts: Prompts):
+
+        self._generate_migration_list()
+
+        plan_confirmed = prompts.confirm(
+            "Above Azure Service Principals will be migrated to UC storage credentials, please review and confirm."
+        )
+        if plan_confirmed is not True:
+            return
+
+        execution_result = []
+        for sp in self._final_sp_list:
+            execution_result.append(self._create_storage_credential(sp))
+
+        results_file = self._installation.save(execution_result, filename=self._output_file)
```

Comment:
Does this results file contain secrets?

Reply:
@nfx Are you sure about this? `AzureServicePrincipal` returned by `WorkspaceClient.storage_credentials.create` suppose to have None `client_secret`. Or is there a config that controls this API behavior that I'm not aware of? I don't believe so. 
But in case in the future bugs introduced to storage_credentials API that may expose client_secret in the response, I can add a guard rail that explicitly set `client_secret` to None before save the migration result.

Reply:
> why do you need to save a secret 
No we are not saving secret

> why do we need to save this file for installation?
We want recording what have been migrated successfully. Especially for those storage credentials that are created successfully, but failed the validation.

> why do we need a secret there, after we did migration?
We don't need and don't store secret after migration.